### PR TITLE
Don't try to fix NSCachedURLResponse's leak on iOS 8 and later

### PR DESCRIPTION
**Pull request by 0xced**
Since iOS 8, we are seeing crash reports with a stack trace identical to the one found on [Cannot track down [NSData getBytes:length:] crash][1].

```
Thread : Crashed: com.apple.CFNetwork.addPersistCacheToStorageDaemon
0  libsystem_platform.dylib       0x3044a208 _platform_memmove$VARIANT$CortexA9 + 160
1  Foundation                     0x22df9167 -[NSData(NSData) getBytes:length:] + 118
2  Foundation                     0x22df9167 -[NSData(NSData) getBytes:length:] + 118
3  Foundation                     0x22e21a1b -[NSData(NSData) replacementObjectForCoder:] + 134
4  Foundation                     0x22dc2aff -[NSXPCEncoder _replaceObject:] + 90
5  Foundation                     0x22e240dd -[NSXPCEncoder _encodeArrayOfObjects:forKey:] + 192
6  Foundation                     0x22e212ff -[NSDictionary(NSDictionary) encodeWithCoder:] + 922
7  Foundation                     0x22dc32c9 -[NSXPCEncoder _encodeObject:] + 604
8  Foundation                     0x22dc379d encodeInvocationArguments + 460
9  Foundation                     0x22dc3455 -[NSXPCEncoder encodeInvocation:] + 360
10 Foundation                     0x22dc32c9 -[NSXPCEncoder _encodeObject:] + 604
11 Foundation                     0x22dc2335 -[NSXPCConnection _sendInvocation:proxyNumber:remoteInterface:withErrorHandler:timeout:userInfo:] + 1860
12 Foundation                     0x22dd2823 -[NSXPCConnection _sendInvocation:proxyNumber:remoteInterface:withErrorHandler:] + 58
13 Foundation                     0x22dd27db -[_NSXPCDistantObjectWithError forwardInvocation:] + 114
14 CoreFoundation                 0x2217e831 ___forwarding___ + 352
15 CoreFoundation                 0x220afb88 _CF_forwarding_prep_0 + 24
16 CFNetwork                      0x21c52ac9 -[NSURLStorage_CacheClient addCachedResponseWithDictionary:key:] + 120
17 CFNetwork                      0x21c21e29 ___ZN12__CFURLCache23CreateAndStoreCacheNodeEP16__CFURLCacheNodePK20_CFCachedURLResponsePK10__CFStringPK13_CFURLRequestPKvbRb_block_invoke + 1576
18 libdispatch.dylib              0x302cf423 _dispatch_call_block_and_release + 10
19 libdispatch.dylib              0x302d95d9 _dispatch_queue_drain$VARIANT$mp + 948
20 libdispatch.dylib              0x302d90a9 _dispatch_queue_invoke$VARIANT$mp + 84
21 libdispatch.dylib              0x302db0d3 _dispatch_root_queue_drain + 330
22 libdispatch.dylib              0x302dc1fb _dispatch_worker_thread3 + 106
23 libsystem_pthread.dylib        0x3044ce25 _pthread_wqthread + 668
```

The crash happens while adding a cached response (frame 16) when accessing some NSData instance which seems bogus (frame 1).

It turns out that RestKit is intentionally over releasing a NSData instance in  [`[NSCachedURLResponse rkData]`][2] as a fix for a leak in the `[NSCachedURLResponse data]` method. This workaround was introduced in commit b4a49c11ffce2954b5252d266516053a0e518bed (pull request #1459).

I can not reproduce this crash but it seems clear to me that the workaround for the leak and the crash are linked. I think a change in iOS 8 somehow exposed this behavior.

  [1]: http://stackoverflow.com/questions/28949799/cannot-track-down-nsdata-getbyteslength-crash
  [2]: https://github.com/RestKit/RestKit/blob/45aee2634cb03f05543ede557b470232af846daf/Code/Network/RKObjectRequestOperation.m#L622-L645

### CODE REVIEW

File: Code/Network/RKObjectRequestOperation.m [hunk 1]
```diff
@@ -623,6 +623,10 @@ @implementation NSCachedURLResponse (RKLeakFix)
 
 - (NSData *)rkData
 {
+    if ([NSProcessInfo instancesRespondToSelector:@selector(operatingSystemVersion)]) {
```
Comment (segiddins) [inline Code/Network/RKObjectRequestOperation.m]: please add a comment as to why this is the correct check

**Comment by 0xced** [comment]
I just force pushed the branch with an explanation.

**Comment by segiddins** [comment]
What effect will this have on OS X? Maybe @blakewatters can chime in about this?

**Comment by 0xced** [comment]
Oops, I did not think about that. Given the comments, I think the original fix did not think about it either. :-/   I just pushed a new commit which applies the workaround on iOS only.

**Comment by 0xced** [comment]
After publishing an app with this potential fix, I still get crashes on iOS 8. I am perplexed.

**Comment by segiddins** [comment]
@0xced safe to close this for now?

**Comment by dl** [comment]
FWIW, I intermittently see the same crash with the same stack trace using v0.24.1 on OSX 10.10+

**Comment by rpranata** [comment]
I dont think this solves the problem. I have the fix in the production but still seeing the crash.

**Comment by jiehu5114** [comment]
NSCachedURLResponse.data  still memory  leak  after iOS8 @0xced

**Pull request closed without merging**